### PR TITLE
refactor file structure to allow for header names occurring in multiple files

### PR DESCRIPTION
**Pull request by hwjsnc**
Also added some documentation for each header to show where it is used (and how).

### CODE REVIEW

File: schemas/cs3/tbl/qs_chapter.ksy [hunk 1]
```diff
@@ -2,6 +2,10 @@ meta:
   id: qschapter # QSChapter
   endian: le
   encoding: UTF-8
+doc: |
+  Monster notebook chapters.
```
Comment (nnguyen259) [inline schemas/cs3/tbl/qs_chapter.ksy]: I think everything related to a notebook (mon, cook, ...) will have this QSChapter, so we will need to change this to reflect it

Comment (hwjsnc) [reply schemas/cs3/tbl/qs_chapter.ksy]: I think we can change the doc as we add those, or should I change it now?

Comment (hwjsnc) [reply schemas/cs3/tbl/qs_chapter.ksy]: Changed it and fixed some other remaining issues.

**Comment by nnguyen259** [comment]
Did you run the new structure through all the supported files? If the results are all good then we can merge this. The other PR will also need to be updated after as well.

**Comment by hwjsnc** [comment]
There were still some typos which are now fixed. The current version compiles to C++ as expected, so it should be good to merge.

**Comment by hwjsnc** [comment]
I think the generation of effects (PR #3) is unaffected by this refactor. I'll update `item` (PR #2) once that's done and merged.

**Pull request merged**